packages/node-core/src/integrations/contextlines.ts

```diff
@@ -62,6 +70,19 @@ function shouldSkipContextLinesForFile(path: string): boolean {
   return false;
 }
 
+/**
+ * Skip frames that we determine to already have been sourcemapped.
+ */
+function shouldSkipContextLinesThatAreAlreadySourceMapped(path: string, frame: StackFrame): boolean {
+  // For non-in-app frames, we skip context lines when we are reasonably certain that the path is already sourcemapped.
+  // For now, we only consider .ts files because they can never appear otherwise in a stackframe, if not already sourcemapped.
+  if (frame.in_app === false && path.endsWith('.ts')) {
```

Comment:
super-l: We could probably match against `/.(c|m)?tsx?/` to cover some more TS file variations. Happy to leave this up to you though

```suggestion
  if (frame.in_app === false && path.endsWith('.ts')) {
```